modules/references/server/controllers/reference.server.controller.js

```diff
@@ -546,3 +546,42 @@ exports.readMine = async function readMine(req, res) {
   }
   return res.status(200).json(reference);
 };
+
+exports.getCount = async function getCount(req, res, next) {
+  const { userTo } = req.query;
+
+  if (!mongoose.Types.ObjectId.isValid(userTo)) {
+    return res
+      .status(400)
+      .send({ message: 'Missing or invalid `userTo` request param' });
+  }
+
+  try {
+    const isSelf = req.user._id.equals(userTo);
+
+    const query = {
+      userTo: new mongoose.Types.ObjectId(userTo),
+    };
+
+    const publicCount = await Reference.find({
+      ...query,
+      public: true,
+    }).count();
+
+    // Include non-public references only when userTo is self
+    const privateCount = isSelf
+      ? await Reference.find({
+          ...query,
+          public: false,
+        }).count()
+      : 0;
+
+    return res.status(200).json({
+      count: privateCount + publicCount,
+      // `hasPending` included only for own profile
+      ...(isSelf ? { hasPending: Boolean(privateCount) } : {}),
```

Comment:
Is it worth returning private and public counts separately for the self profile (since we already got these values from the DB)? Then, `hasPending` can be derived from `privateCount` on the client. Additionally, the private count may be useful in the future.

Reply:
My thinking was that the exact count for pending ones isn't useful in client and I'd prefer just returning data in format we actually need it at the client side.

> Additionally, the private count may be useful in the future.

We can add it back at that point if it becomes useful.

What do you think?

Reply:
sure